api/src/api/healthCheckRouter.test.ts

```diff
@@ -0,0 +1,49 @@
+import { healthCheckRouterFactory } from "@api/healthCheckRouter";
+import type { HealthCheckMethod } from "@domain/types";
+import { setupExpress } from "@libs/express";
+import type { Express } from "express";
+import request from "supertest";
+
+const mockHealthCheckMethod: HealthCheckMethod = async () => {
+  return new Promise((resolve) => {
+    resolve({
+      success: true,
+      data: {
+        message: "Alive",
+      },
+    });
+  });
+};
+
+describe("healthCheckRouter", () => {
+  let app: Express;
+
+  beforeEach(async () => {
+    app = setupExpress(false);
+    app.use(healthCheckRouterFactory(new Map<string, HealthCheckMethod>([["mock", mockHealthCheckMethod]])));
+  });
+
+  afterAll(async () => {
+    await new Promise((resolve) => {
+      return setTimeout(resolve, 500);
+    });
+  });
+
+  describe("GET /", () => {
+    it("should return a successful response for the server's self check", async () => {
+      const response = await request(app).get("/");
+      expect(response.status).toBe(200);
+      expect(response.body.success).toBeTruthy();
```

Comment:
[pebble] I think we should be using `toBe(true)` instead of` toBeTruthy()` because we explicitly want to check for a `true` and `toBeTruthy()` evaluates to true when there is any value that is not `false`, `0`, `''`, `null`, `undefined`, and `NaN` and that is not what we are testing in this case?